lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -1536,6 +1536,18 @@ bool SROA_HLSL::performScalarRepl(Function &F, DxilTypeSystem &typeSys) {
   for (auto A : AllocaList)
     HLModule::MergeGepUse(A);
 
+  // Make sure big alloca split first.
+  // This will simplify memcpy check between part of big alloca and small
+  // alloca. Big alloca will be split to smaller piece first, when process the
+  // alloca, it will be alloca flattened from big alloca instead of a GEP of big
+  // alloca.
+  auto size_cmp = [&](AllocaInst *a0, AllocaInst *a1) -> bool {
```

Comment:
Better to be explicit about what you are capturing (i.e. `&DL`) and you can take the AllocaInst as a const pointer.